### PR TITLE
STUD-331: Add earnings claim modal and required endpoints

### PR DESCRIPTION
**Pull request by escobarjonatan**
https://github.com/user-attachments/assets/828738f7-7cd4-48ea-bde0-ef8ba0266613

### CODE REVIEW

File: apps/studio/src/modules/wallet/types.ts [hunk 1]
```diff
@@ -3,3 +3,58 @@ export interface WalletState {
...
+  readonly amountCborHex: string;
+  readonly earnings: Earning[];
+  readonly totalClaimed: number;
+};
+
+export type PostEarningsRequest = {
+  readonly changeAddress: string;
+  readonly utxoCborHexList: string[];
+  readonly walletAddress: string;
+};
```
Comment (scandycuz) [inline apps/studio/src/modules/wallet/types.ts]: Was defining these as types instead of interfaces intentional?

Comment (escobarjonatan) [reply apps/studio/src/modules/wallet/types.ts]: negative, I switch between types and interfaces in different projects. made them interface for consistency here

Comment (scandycuz) [reply apps/studio/src/modules/wallet/types.ts]: I was thinking the `GetEarningsResponse` and `PostEarningsRequest` types would be more consistent to have as interfaces.

**Review by scandycuz — APPROVED** [review]
Had a minor comment, but LGTM!

**Pull request merged**